test/functional/feature_evm_proxy.py

```diff
@@ -0,0 +1,281 @@
+        tx = compiled.constructor().build_transaction(
+            {
+                "chainId": node.w3.eth.chain_id,
+                "nonce": node.w3.eth.get_transaction_count(self.evm_key_pair.address),
+                "maxFeePerGas": 10_000_000_000,
+                "maxPriorityFeePerGas": 1_500_000_000,
+                "gas": 1_000_000,
+            }
+        )
+        signed = node.w3.eth.account.sign_transaction(tx, self.evm_key_pair.privkey)
+        hash = node.w3.eth.send_raw_transaction(signed.rawTransaction)
+        node.generate(1)
+        receipt = node.w3.eth.wait_for_transaction_receipt(hash)
+        implementation_contract = node.w3.eth.contract(
+            address=receipt["contractAddress"], abi=implementation_abi
+        )
+        return implementation_contract
+
+    def should_deploy_proxy_smart_contract(
+        self, implementation_contract: web3Contract
+    ) -> web3Contract:
+        node = self.nodes[0]
+        abi, bytecode = EVMContract.from_file(
+            "SimpleProxy.sol", "SimpleERC1967Proxy"
+        ).compile()
+        compiled = node.w3.eth.contract(abi=abi, bytecode=bytecode)
+
+        tx = compiled.constructor(
+            implementation_contract.address,
+            implementation_contract.encodeABI(
+                "initialize", [self.evm_key_pair.address, 10]
+            ),
+        ).build_transaction(
+            {
+                "chainId": node.w3.eth.chain_id,
+                "nonce": node.w3.eth.get_transaction_count(self.evm_key_pair.address),
+                "maxFeePerGas": 10_000_000_000,
+                "maxPriorityFeePerGas": 1_500_000_000,
+                "gas": 1_000_000,
+            }
+        )
+        signed = node.w3.eth.account.sign_transaction(tx, self.evm_key_pair.privkey)
+        hash = node.w3.eth.send_raw_transaction(signed.rawTransaction)
+        node.generate(1)
+
+        receipt = node.w3.eth.wait_for_transaction_receipt(hash)
+        proxy_contract = node.w3.eth.contract(
+            address=receipt["contractAddress"], abi=implementation_contract.abi
+        )
+        assert_equal(proxy_contract.functions.randomVar().call(), 10)
+        return proxy_contract
+
+    def should_call_function_in_implementation_contract(
+        self, proxy_contract: web3Contract
+    ):
+        node = self.nodes[0]
+        tx = proxy_contract.functions.setRandomVar(20).build_transaction(
+            {
+                "chainId": node.w3.eth.chain_id,
+                "nonce": node.w3.eth.get_transaction_count(self.evm_key_pair.address),
+                "gas": 1_000_000,
+            }
+        )
+        signed = node.w3.eth.account.sign_transaction(tx, self.evm_key_pair.privkey)
+        hash = node.w3.eth.send_raw_transaction(signed.rawTransaction)
+        node.generate(1)
+        receipt = node.w3.eth.wait_for_transaction_receipt(hash)
+        assert_equal(receipt["status"], 1)
+        assert_equal(proxy_contract.functions.randomVar().call(), 20)
```

Comment:
since its named `should_call_function_in_implementation_contract`
missing the call to implementation contract??

Reply:
Oh, what I mean is proxy delegate call to implementation contract. Changed the name of the test